### ISSUE TITLE
Add ability for custom screens on kiosks

### ISSUE THREAD

**Issue by Jchuchla** [jeffpiazza/derbynet#301]
There's often other announcements and such that happen during the pinewood derby.  It would be nice to be able to create custom kiosk screens to support this.  For example today we wanted to show information for future events, QR codes for event signups, etc...

**Comment by majora15** [comment]
You can do this it just requires creating the page in a specific location. You do need to make sure a little bit of java code is added to the page, so it checks in with the server to make sure the kiosk is displaying the correct information.

So for location and what needs to be on the page, I can get that to you in a little bit.

**Comment by majora15** [comment]
/var/www/html/derbynet/kiosks

These are the three lines that should exist in the custom kiosk page:

<?php require('inc/kiosk-poller.inc'); ?>
<?php require('inc/stylesheet.inc'); ?>

<?php require_once('inc/ajax-failure.inc'); ?>

example: 
`<!DOCTYPE html>
<html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8"/>
<title>Welcome</title>
<script type="text/javascript" src="js/jquery.js"></script>
<?php require('inc/kiosk-poller.inc'); ?>
<?php require('inc/stylesheet.inc'); ?>
<link rel="stylesheet" type="text/css" href="css/kiosks.css"/>
<style>
.full_page_center img {
max-width: 100%;
max-height: 80vh;
}
</style>
</head>
<body>
<div class="full_page_center">
<p class="kiosk_title">Cub Scout - Pinewood Derby</p>
<img src="image.php/emblem"/>
</div>
<p class="kiosk_title">Welcome Pack</p>

<?php require_once('inc/ajax-failure.inc'); ?>

</body>
</html>
`

**Comment by Jchuchla** [comment]
Thanks.  That will get it done for technical users like me.  Though when I wrote up the request, I had more of a no-code gui based implementation in mind.  Like the ability to create new pages from the GUI, and populate it with simple text or fullscreen image.

**Comment by majora15** [comment]
Understandable.

**Comment by jeffpiazza** [comment]
There's nascent support for looking for .kiosk files in a user-managed directory (.../local/kiosks/), but it needs to be more fully developed to allow user-provided static slides or images.  Probably some docs on how kiosk pages work (mainly the polling to see if something else should display) and how to implement your own, for those inclined to do so.

I don't foresee providing a native editor for building your own page through the web interface.  Maybe there's a case to be made that a generic announcement page with customizable strings would serve a purpose, but at the moment I'm not inclined to go that way.

**Comment by Jchuchla** [comment]
I agree that it's not worth a lot of dev time to make an editor or anything like that.  Perhaps just a big text field that the user can enter their own stuff, allowing HTML.  Insert that whole block as-is into the page's HTML.  If it's marked up with tags, great, if it's not, then it just shows as plain text with whatever default CSS applies.
Maybe an upload button to upload image files to the server that the user can then reference via their own HTML.

**Comment by robertrightmire-debug** [comment]
I was wondering about a custom image for the hosted option, and found this page. would it be overly difficult to have a link to a hosted image as an option? users could use any number of online imaging sites even temporary hosting and just have a kiosk with a place to paste in the url?
